### ISSUE TITLE
[Feature Request] Avoid multiple notifications with "Single notification after inactivity" or "Single notification per chat"

### ISSUE THREAD

**Issue by KXPN** [telegramdesktop/tdesktop#8444]
### Is your feature request related to a problem?
We all have this typical friend which, instead of using new lines, sends multiple messages for a single simple thing, so we end up with 10 notifications for a single subject. Multiply that for all your friends that act like that, and all the groups you're in, and you'll get a lot of notifications which end up forcing you to disable notifications for a while just so you don't hear them anymore

### Describe the solution you'd like
As I see it, there are two ways of fixing this:

1. **Notify after `X` seconds of inactivity** option, which allows the user to select the amount of seconds of inactivity, after which the notification should be displayed. Example: User `A` sets the option to 10 seconds. User `B` sends a message to `A`. `A` doesn't get notified, but will be in `X` seconds. Three seconds pass. `A` will be notified in `X - 3` seconds. `B` sends another message to `A`. `A` will be notified in `X` seconds (counter starts again). Finally, 10 seconds pass since last message (13 since first message), since no other message was received, `A` gets a single notification about two messages from `B`, instead of two notifications in real time as it happens now
2. **Notify only once per chat** option, which allows the user to be notified only once per chat since it was last read. Example: User `A` activates the option. User `B` sends a message to `A`. `A` gets a notification. `B` sends another message to `A`. `A` won't get a sound notification anymore, but the counter will keep counting. `B`sends more messages to `A`. `A` still won't get notified. `A` opens chat with `B`, reads all unread messages and closes the chat. `B` sends another message to `A`. `A` gets notified since all messages were read

For both cases, `B` could also be a group or a channel. It would be cool to set both chat-wise, instead of as a general rule. But the general rule approach might be really good anyway

### Describe alternatives you've considered
- Disable notifications, but this way one won't get any notifications at all
- Mute people, groups or channels that send a lot of things, but one might not want that

### Additional context
No other chat I've tried offers a configuration like this, and it will be a really nice to have. It can help people get less stress from a lot of chat and have a better experience overall. I would really like to have both, probably using option 2 for groups and 1 for people

**Comment by ilya-fedin** [comment]
#429
#7618
